### PR TITLE
Link Godot Ideas repository as an alternative for GIP

### LINKED ISSUE #39: Features and Enhancements should be treated separately or questions be rephrased
Relates to #10.

Noticed some issue with people not filling templates properly.

There should be two kinds of templates in GIP: one handling new features, and another enhancing existing ones, that way people could avoid filling out redundant questions. The template itself is named as `Feature / Enhancement Proposal` when you attempt to create one, so makes sense to split those just like in the main repository.

I propose modifying those templates like the following:

# Feature template
~~**Describe the project  you are working on**~~
**Describe the vision/end goal you are trying to achieve**
* not sure how to phrase it better but some people might not want to describe their own projects for whatever reason (they even might be under NDA, who knows).

~~**Describe how this enhancement will help your project:**~~
**Describe how this feature will help to achieve your goal:**

**Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:**

**Describe implementation detail for your proposal (in code), if possible:**

~~**If this feature will not be used often, can it be worked around with a few lines of script?:**~~
* this feels a bit more like a hint/guide rather than an actual question to fill.
* this also pertains to enhancement rather than feature proposal.
* "working around" makes sense for existing features.
* the feature should be used relatively often within domain!

**Is there a reason why this should be core and not an add-on in the asset library?:**

# Enhancement template
~~**Describe the project you are working on**~~
**Describe enhancement of the existing feature** (https://github.com/godotengine/godot-proposals/issues/39#issuecomment-528819736)
* most people refer to this as "irrelevant" when filling out blanks #37, #35.
* some just learn stuff #28.

~~**Describe how this feature / enhancement will help your project:**~~
**Describe how this enhancement will help to achieve your goal:**

~~**Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:**~~
* somewhat irrelevant because the ground work is already done.

**Describe implementation detail for your proposal (in code), if possible:**

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**

~~**Is there a reason why this should be core and not an add-on in the asset library?:**~~
* the feature is already there in the core #37, #36

---

A few other questions could be added for each.

### LINKED ISSUE #47: Allow to discuss and present ideas freely for core and/or plugin development
## The bar is too high just to spark ideas here

Continuing an off-topic discussion from https://github.com/godotengine/godot-proposals/issues/39#issuecomment-529095104.

*Note*: I'm personally against segregation but there seems to be no consensus with maintainers so I'm just seeking alternatives.

## Example repository
[godot-ideas](https://github.com/godot-extended-libraries/godot-ideas)

It mimics the process of `godot-proposals` without too much restrictions, and users are pointed to this repository if they feel like they could get approval from core developers, else it would be a nice place to discuss stuff without imposing too much rules.

## Problem
We've established previously that proposals could be classified as:

* Nice-to-have
* Need-to-have (aka must-have)

The "Nice-to-have" type of proposals, based on the official guide and feedback, are going to be closed eventually without them being considered for review, because allegedly they don't fulfill the actual needs and fix problems that users are dealing with on day-to-day basis.

While I completely understand the amount of "low-quality" proposals that are going to spawn if we waive the restriction of having to fill out every single question, the need to express ideas and be heard still exists, and we'd rather avoid dealing with frustrated users.

The listed community channels aren't as functional or reliable as they could be, see https://github.com/godotengine/godot-proposals/issues/47#issuecomment-529203868.

### Human factor 🙂

1. I guess people wouldn't like their proposals to be "closed" even if they can be reopened by maintainers if deemed satisfactory or needed, there's enough of examples how this makes people discouraged and rather frustrated.

2. The feeling of guilt that people feel starting random and general-purpose discussions by having to type "wall of text" godotengine/godot#19486. This can be avoided with dedicated place for that.

## Proposal
There should be an **_official_** repository for people to allow to discuss their ideas there. Maintainers are free to "ignore" these ideas so they don't really have to put much effort by having to read and understand them. Don't feel like users are **demanding** you to make it happen. Moreover, such repository would be a place for people to ask the questions like:

* Is it possible to make this as a plugin or should I go make a proposal to `godot-proposals`?
* I'm not sure whether this feature is needed at all, but perhaps it will in the future?

[Controversial topics](https://github.com/godotengine/godot/issues?q=is%3Aopen+is%3Aissue+label%3Adiscussion+label%3A%22feature+proposal%22+sort%3Acomments-desc) would be a perfect fit for this.

Such repository wouldn't necessary be targeted for Godot core, but a place for people to kind of request independent developers to write modules/plugins for them, but it needs to be centralized. An author is free to open and close an issue on his own.

Also see alternative approach/reasoning in https://github.com/godotengine/godot-proposals/issues/47#issuecomment-529388737.

<details>
  <summary>A list of proposals with mentioned issues or criteria</summary>

(if not relevant, see first edits in mentioned issues)
* #58 - plugin-candidate.
* #55 - just an idea, proposal template disregarded.
* #63 - the author is not sure whether it could be made as plugin or could be part of the core, avoiding presenting more ideas + template ignored.
* #65 - nice-to-have, possibly plugin-candidate, met with disapproval which could otherwise be welcomed.

</details>

### LINKED ISSUE #91: Add a godot-ideas referral to the issue template
**Describe the project you are working on:**
Managing community expectations for the Godot-Proposals repository.

**Describe how this feature / enhancement will help your project:**
I think we should add a blurb in the `godot-proposals` issue template that redirects people to the new `godot-extended-libraries/godot-ideas` repository if they do not feel as though they can adequately answer the questions, or if they already know that their proposal is not applicable to the engine development itself.

The reason this would help, aside from things already brought up in #47, is that it drives users awareness of this space up.

**Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:**

See below.

**Describe implementation detail for your proposal (in code), if possible:**

Just add a comment to the top of the message, exactly like the one in `godotengine/godot` that tells people they can go to `godot-extended-libraries/godot-ideas` instead.

> \<!--If you feel that you cannot provide highly detailed instructions with the proposal below, consider creating a more simple, open-ended Issue in the unofficial, community-maintained [godot-extended-libraries/godot-ideas repository](https://github.com/godot-extended-libraries/godot-ideas/issues/new)--\>

Something like that.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**

Relates to the GitHub repository, not Godot itself.

**Is there a reason why this should be core and not an add-on in the asset library?:**

See above.

### LINKED ISSUE #779: The proposal process doesn't seem very friendly, especially to the new person
I've noticed a problem with this repository, a problem which seems like it could push away a lot of people, and honestly one that's been on my mind for a while.

I've seen quite a few times where the response to a proposal seems like someone trying to figure out how to avoid implementing the feature... This isn't normally a problem, mind you (although I do think that the usual "you can do this with the current nodes and a script" isn't very helpful to people). What makes it a problem for me is that it often comes at the expense of not considering what we *could* reasonably do to help someone with what they want.

I should put a disclaimer: I don't frequent this repository anymore, but I do still read its issues from time-to-time (I have a habit of not cleaning out lists such as subscription lists). And I've seen one too many times where it feels like the responses given end up feeling like someone trying to say "no, and that's final, here's what you can do" rather than "no, but here's what we could do instead".

Don't get me wrong, I'm not trying to say that we can't say "no" in the way that we currently do -- but with how much we push the work onto the average proposal maker ("you can do this in GDScript"), rather than first considering what options we could implement to help them further their goal, it feels much like a place where ideas have to be accepted immediately or be rejected forever.

It ends up being, I think, a place where you have to already know how Godot is designed in order to be able to suggest ideas... and I'd have a hard time believing that the average person could figure that out that, given that I don't even have the confidence in whether I've done so myself, even after examining Godot's internals.

Again, I don't frequent this repository anymore, so please take this complaint as you will. Also, I apologize if it seems like a mess of thoughts, as I'm not always good at articulating my thoughts. I've just... ended up incredibly frustrated trying to think up ideas that I genuinely think would fit into Godot by default, only to feel as though they'd end up rejected because of how the proposal process works -- and so, I don't ever end up feeling like I can contribute good ideas because I'm not sure which ones are good and which ones aren't.

### PR DESCRIPTION
**Pull request by Xrayez**
Resolves #39.
Resolves #47.
Closes #91, as suggested by @Calinou in https://github.com/godotengine/godot-proposals/issues/91#issuecomment-688850693.
Closes #779, similar suggestion by @aaronfranke in https://github.com/godotengine/godot-proposals/issues/779#issuecomment-625595439, with the 👍 from the OP.

Currently, GIP has a requirement that proposals must describe a concrete use case to justify making a proposal:

> All proposals must be linked to a substantive use-case. In justifying your proposal, it is not enough to say it would be "nice" or "helpful". Use the template to show how Godot is not currently meeting your needs and then explain how your proposal will meet a particular need.

This pull request amends the third rule for submitting a proposal to provide an alternative way to present and discuss ideas in the [`Godot Ideas`](https://github.com/godot-extended-libraries/godot-ideas) repository (unofficial) in case people do not have any concrete use cases to justify the proposal.

### CODE REVIEW

**Review by Calinou — APPROVED** [review]

**Comment by aaronfranke** [comment]
@Xrayez Perhaps this should be a sub-point? The syntax looks like this:

```md
1. One
    * Sub-point of One
2. Two
    * Sub-point of Two
```

**Comment by Xrayez** [comment]
Yes, this improves visibility/readability indeed. 🙂

**Review by aaronfranke — APPROVED** [review]

**Comment by Xrayez** [comment]
As kindly pointed out by Calinou in https://github.com/godotengine/godot-proposals/issues/1634#issuecomment-706564283, I've stumbled upon ["Before creating an RFC"](https://github.com/rust-lang/rfcs#before-creating-an-rfc) section in the Rust's RFC repository. I'll quote the entire thing for the purpose of recording this, and put some emphasis with **bold**:

> A hastily-proposed RFC can hurt its chances of acceptance. Low quality proposals, proposals for previously-rejected features, or those that don't fit into the near-term roadmap, may be quickly rejected, which **can be demotivating for the unprepared contributor**. **Laying some groundwork ahead of the RFC can make the process smoother**.
>
> Although there is no single way to prepare for submitting an RFC, it is generally a good idea to pursue feedback from other project developers beforehand, to ascertain that the RFC may be desirable; having a consistent impact on the project requires concerted effort toward consensus-building.
>
> The most common preparations for writing and submitting an RFC include **talking the idea over** on our official Discord server, discussing the topic on our developer discussion forum, and occasionally **posting "pre-RFCs" on the developer forum**. You may file issues on this repo for discussion, but these are not actively looked at by the teams.
>
> As a rule of thumb, receiving encouraging feedback from long-standing project developers, and particularly members of the relevant sub-team is a good indication that the RFC is worth pursuing.

That's one of the functions behind the proposed Godot Ideas repository, to give people ability to pre-evaluate whether their ideas are worth pursuing for the engine development. If not, then those ideas are certainly plugin candidates, and that's where GIP is not suitable for those kind of proposals, am I right?

Godot Ideas is not any different from other Godot Engine *community* channels. I just don't see any other solution. There are definitely other alternatives from the GIP side:
1. Allow people to create non-proposal, discussion issues. Those don't have to be evaluated by the core developers.
2. Amend some GIP rules to be less restricting. But as Godot keeps growing, I suspect that we'll get even more strict rules for submitting proposals, just like with the Rust's RFC above.

So, is there a reason why this PR has not been merged yet? 😛

**Comment by Xrayez** [comment]
@akien-mga @clayjohn sorry for disturbing, but I'd like to receive your feedback on this (even negative), or at least say that you're aware of this and will look into this later, it's been almost 3 months since I created this PR.

Thanks!

**Comment by Xrayez** [comment]
We have enabled GitHub's "Discussions" feature in Godot Ideas repository: https://github.com/godot-extended-libraries/godot-ideas/discussions, so perhaps that's something Godot Proposals can enable as well, I believe this would solve most of our inquiries.

This way, there's no need for open or close issues, and those discussions don't have to be evaluated by the core developers.

**Comment by Xrayez** [comment]
I've created another meta proposal #2069 which might as well supersede this pull request, which would also pretty much solve most of the issues we've been discussing throughout the past year...

**Comment by Xrayez** [comment]
What's the status of this PR? Is it safe to merge?

**Comment by clayjohn** [comment]
I think Calinou and AaronFranke should review again since there have been a few force pushes since they initially approved.

**Review by aaronfranke — APPROVED** [review]

**Comment by Xrayez** [comment]
The only reason why I had to force-push was to resolve merge conflicts, as usual!

**Comment by Xrayez** [comment]
I'm sorry but do we have consensus or not? It's been 2 months from our last interaction. Lets just merge this and see how it goes?

**Comment by clayjohn** [comment]
Merging now. 

As a heads up, there has been a lot of discussion again about using Github discussions (as you originally proposed I believe). If we do end up moving to discussions, then this PR will likely need to be reverted.

**Comment by Xrayez** [comment]
> If we do end up moving to discussions, then this PR will likely need to be reverted.

Yeah, that's a fair decision, thanks!

**Pull request merged**
